test/functional/wallet_listtransactions.py

```diff
@@ -210,5 +212,57 @@ def get_unconfirmed_utxo_entry(node, txid_to_match):
         assert_equal(self.nodes[0].gettransaction(txid_3b)["bip125-replaceable"], "no")
         assert_equal(self.nodes[0].gettransaction(txid_4)["bip125-replaceable"], "unknown")
 
+    def create_and_send_transaction(self, utxo, address, amt, feeRate):
+        psbtx = self.nodes[0].walletcreatefundedpsbt([{"txid": utxo['txid'], "vout": utxo['vout']}],
+                                                      {address: amt},
+                                                      0,
+                                                      {"replaceable":True, "feeRate":feeRate})['psbt']
+        signed_tx = self.nodes[0].walletprocesspsbt(psbtx)['psbt']
+        final_tx = self.nodes[0].finalizepsbt(signed_tx)['hex']
+        return self.nodes[0].sendrawtransaction(final_tx)
+
+    def test_listtransactions_display_in_mempool(self):
+        self.log.info('Testing that listtransactions correctly displays whether a transaction is in the mempool')
+        utxo = self.nodes[0].listunspent()[0]
+        address = self.nodes[0].getnewaddress()
+
+        tx1_id = self.create_and_send_transaction(utxo, address, 0.1, 0.001)
+
+        new_txs = self.nodes[0].listtransactions(count=2)
+        for tx in new_txs:
+            assert_equal(tx['txid'], tx1_id)
+            assert_equal(tx['in_mempool'], True)
+
+        tx2_id = self.create_and_send_transaction(utxo, address, 0.1, 0.002)
+
+        new_txs = self.nodes[0].listtransactions(count=4)
+        for i in range(2):
+            assert_equal(new_txs[i]['txid'], tx1_id)
+            assert_equal(new_txs[i]['in_mempool'], False)
+
+        for i in range(2, 4):
+            assert_equal(new_txs[i]['txid'], tx2_id)
+            assert_equal(new_txs[i]['in_mempool'], True)
+
+    def test_gettransaction_display_in_mempool(self):
+        self.log.info('Testing that gettransaction correctly displays whether a transaction is in the mempool')
+        utxo = self.nodes[0].listunspent()[0]
+        address = self.nodes[0].getnewaddress()
+
+        tx1_id = self.create_and_send_transaction(utxo, address, 0.1, 0.001)
+
+        tx1 = self.nodes[0].gettransaction(tx1_id)
+        assert_equal(tx1['txid'], tx1_id)
+        assert_equal(tx1['in_mempool'], True)
+
+        tx2_id = self.create_and_send_transaction(utxo, address, 0.1, 0.002)
+        tx1 = self.nodes[0].gettransaction(tx1_id)
+        tx2 = self.nodes[0].gettransaction(tx2_id)
+        assert_equal(tx1['txid'], tx1_id)
+        assert_equal(tx1['in_mempool'], False)
+        assert_equal(tx2['txid'], tx2_id)
+        assert_equal(tx2['in_mempool'], True)
```

Comment:
It would be good to also test for the absence of the `in_mempool` field when the transaction has more than zero confirmations, e.g. somethnig like

```python
+        node.generate(1)
+        self.sync_all()
+        tx2 = node.gettransaction(tx2_id)
+        assert_equal(tx2["confirmations"], 1)
+        assert "in_mempool" not in tx2  # only present if txn is unconfirmed
```